src/lib/utils/loadstor.h

```diff
@@ -169,9 +169,37 @@ constexpr bool native_endianness_is_unknown() {
  * Models a type that can be loaded/stored from/to a byte range.
  */
 template <typename T>
-concept unsigned_integralish = std::unsigned_integral<T> || concepts::unsigned_integral_strong_type<T> ||
+concept unsigned_integralish = std::unsigned_integral<strong_type_wrapped_type<T>> ||
                                (std::is_enum_v<T> && std::unsigned_integral<std::underlying_type_t<T>>);
 
+template <unsigned_integralish T>
+using wrapped_type = decltype([] {
+   if constexpr(std::is_enum_v<T>) {
+      return std::underlying_type_t<T>{};
+   } else {
+      return Botan::strong_type_wrapped_type<T>{};
+   }
+}());
+
+template <unsigned_integralish InT>
+constexpr auto unwrap_strong_type_or_enum(InT t) {
+   if constexpr(std::is_enum_v<InT>) {
+      // TODO: C++23: use std::to_underlying(in) instead
+      return static_cast<std::underlying_type_t<InT>>(t);
+   } else {
+      return Botan::unwrap_strong_type(t);
+   }
+}
+
+template <unsigned_integralish OutT, std::unsigned_integral T>
+constexpr auto wrap_strong_type_or_enum(T t) {
+   if constexpr(std::is_enum_v<OutT>) {
+      return static_cast<OutT>(t);
+   } else {
+      return Botan::wrap_strong_type<OutT>(t);
+   }
+}
```

Comment:
This is an extra layer of abstraction to make `load/store` work with integer-enums as well. I feel that this would be useful for the other usage locations as well, so in theory we could also promote that into the implementation in `strong_type.h`. As a result, the `checked_cast_to` would then automagically work with enums as well.